### PR TITLE
LPS-146479 [CP 2.0] Create graphql request for Liferay staff to view all projects

### PR DESCRIPTION
**Pull request by joaosvassis**
# How it was developed?
   - I used Graphql to query all the projects and check if the user is an administrator.

**Query projects**
query_projetos
**Query administrator** 
query_administrator

# What has been affected by the development?
   - I implemented a roleBriefs field in the getUserAccount query.
   
  queries_lps

# Test Plan
  **Test 1**
  - Log in as an administrator user
  - Add 2 or more projects
  - Associate fewer projects than the total number of projects
  - Go to the home page, make sure the user is viewing all projects, regardless of which one is associated
 
**Test 2**
  - Log in with a user that is not an administrator
  - Add 1 or more projects
  - Associate projects to the user
  - Go to the home page, make sure the user is viewing the projects he is associated with

### CODE REVIEW

**Review by iagochaves — COMMENTED** [review]
Looks good! :+1:

**Review by Lucas2691 — COMMENTED** [review]
Looks good!

**Review by joanagrego — APPROVED** [review]
Great job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Home/index.js [hunk 1]
```diff
@@ -35,76 +38,96 @@ const getStatus = (slaCurrent, slaFuture) => {
...
 
+const getKoroneikiFilter = (accounts) => {
+	return accounts
+		?.map(
+			({externalReferenceCode}, index, {length: totalAccounts}) =>
+				`accountKey eq '${externalReferenceCode}'${
+					index + 1 < totalAccounts ? ' or' : ''
+				}`
+		)
+		.join(' ');
```
Comment (igordmf) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Home/index.js:49]: I think that .map followed by .join could be an reduce:


```suggestion
		?.reduce((acc, {externalReferenceCode}, index, {length: totalAccounts}) => `${acc}accountKey eq '${externalReferenceCode}'${index + 1 < totalAccounts && ' or '}`, '');
```

And just one iteration will be done. What do you think?

**Review by igordmf — COMMENTED** [review]
Gob job Joao!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Home/index.js (hunk 1 above)
Comment (joaosvassis) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Home/index.js:49]: I find the change interesting

**Review by iagochaves — APPROVED** [review]

**Review by Lucas2691 — APPROVED** [review]
Nice Work!

**Review by thaynam — APPROVED** [review]
Good job, João!

**Review by igordmf — APPROVED** [review]
Nice job!

**Review by claricebrasil — APPROVED** [review]
Nice job!

**Comment by jairmedeiros** [comment]
@kevenleone, it's our time!!

**Review by jairmedeiros — APPROVED** [review]
Looks good to me!!!!!!! 🚀

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 2]
```diff
@@ -301,6 +301,17 @@ export const getListTypeDefinitions = gql`
 	}
 `;
 
+export const getAccounts = gql`
+	query getAccounts {
+		accounts {
+			items {
+				name
+				externalReferenceCode
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:309]: Alphabetical Order

```suggestion
				externalReferenceCode
				name
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 3]
```diff
@@ -313,6 +324,9 @@ export const getUserAccount = gql`
 					name
 				}
 			}
+			roleBriefs {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:327]: roleBriefs should be placed in alphabetical order, below name

**Review by kevenleone — CHANGES_REQUESTED** [review]
@joaosvassis can you order the properties from query?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js (hunk 2 above)
Comment (joaosvassis) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js]: Ok, thanks

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js (hunk 3 above)
Comment (joaosvassis) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js]: Ok, thanks

**Comment by jairmedeiros** [comment]
ci:foward

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**